### PR TITLE
Add guide to migrating existing codes

### PR DESCRIPTION
**Pull request by rchodava**

### CODE REVIEW

File: docs/api-clients/access-codes/list-unmanaged-access-codes.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+description: Get all Unmanaged Access Codes of a Device by its ID
+---
+
+The [guide to migrating existing access codes](../device-guides/migrating-existing-codes.md) provides an overview of the difference between
+unmanaged and managed access codes.
```
Comment (razor-x) [inline docs/api-clients/access-codes/list-unmanaged-access-codes.md:6]:
```suggestion
managed and unmanaged access codes.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+---
+description: Learn how to migrate existing lock access codes into Seam
+---
+
+# Overview
+
+Prior to using Seam to manage your devices, you may have used another lock management system to manage
+the access codes on your devices. Where possible, we will help you keep any existing access codes on devices, and
+transition those codes to ones managed by Seam.
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:9]: Maybe 🤔 
```suggestion
transition those codes to ones managed by your Seam workspace.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+Where it isn't possible to transition existing codes to be Seam-managed, we will allow you to get a list of
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:11]:
```suggestion
If migrating existing codes into your Seam workspace is not possible, you can still get a list of
```

File: docs/device-guides/migrating-existing-codes.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+these existing codes, and replace them with new Seam-managed access codes.
+
+# Unmanaged codes
+
+Seam differentiates between managed codes and unmanaged codes. When you create an access code on a device in Seam,
+it is created as a managed code. Codes that exist on a device that were not created on Seam are considered unmanaged
+codes. We strictly limit the operations that can be performed on unmanaged codes to:
+
+- Viewing a list of the unmanaged codes on a device
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:20]:
```suggestion
- Viewing a list of the unmanaged codes on a device.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+- Converting an unmanaged code into a managed code
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:21]:
```suggestion
- Converting an unmanaged code into a managed code.
- Deleting an unmanaged code from the device.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+{% hint style="info" %}
+Strictly speaking, unmanaged codes are any codes on a device that were not created by the current workspace. So for example,
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:24]:
```suggestion
Strictly speaking, unmanaged codes are any codes on a device that were not created by the current workspace. For example,
```

File: docs/device-guides/migrating-existing-codes.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+if you are within _Workspace 1_ and created codes _Code 1_ and _Code 2_, those two codes would be considered unmanaged if you
+were looking at the same device in _Workspace 2_. Therefore, unmanaged codes include both codes not managed by Seam at all, and
+codes that might be managed by a different workspace within Seam.
+{% endhint %}
+
+# Viewing the unmanaged codes on a device
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:30]:
```suggestion
# Viewing unmanaged codes on a device
```

File: docs/device-guides/migrating-existing-codes.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+# Overview
+
+Prior to using Seam to manage your devices, you may have used another lock management system to manage
+the access codes on your devices. Where possible, we will help you keep any existing access codes on devices, and
+transition those codes to ones managed by Seam.
+
+Where it isn't possible to transition existing codes to be Seam-managed, we will allow you to get a list of
+these existing codes, and replace them with new Seam-managed access codes.
+
+# Unmanaged codes
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:14]:
```suggestion
# Unmanaged access codes
```

File: docs/device-guides/migrating-existing-codes.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+Seam differentiates between managed codes and unmanaged codes. When you create an access code on a device in Seam,
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:16]:
```suggestion
Seam differentiates between managed access codes and unmanaged access  codes. When you create an access code on a device in Seam,
```

File: docs/device-guides/migrating-existing-codes.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+---
+
+# Overview
+
+Prior to using Seam to manage your devices, you may have used another lock management system to manage
+the access codes on your devices. Where possible, we will help you keep any existing access codes on devices, and
+transition those codes to ones managed by Seam.
+
+Where it isn't possible to transition existing codes to be Seam-managed, we will allow you to get a list of
+these existing codes, and replace them with new Seam-managed access codes.
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:12]:
```suggestion
these existing codes, and replace them with new access codes managed by your Seam workspace.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+}
+```
+
+{% endswagger-response %}
+
+{% endswagger %}
+
+# Converting unmanaged codes into managed codes
+
+You can generally convert unmanaged access codes into managed ones using a `POST /access_codes/unmanaged/update` request,
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:64]:
```suggestion
You can convert unmanaged access codes into managed ones using a `POST /access_codes/unmanaged/update` request,
```

File: docs/device-guides/migrating-existing-codes.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+and passing in a JSON body similar to:
+
+```js
+{
+    "access_code_id": "xxxx-xxxx-xxxx-xxxx",
+    "is_managed": true
+}
+```
+
+Note that the request will complete immediately but the conversion is an asynchronous process that happens in the background.
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:74]:
```suggestion
The request will return immediately, but the conversion is an asynchronous process that completes in the background.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+You can listen to the `access_code.unmanaged.converted_to_managed` and `failed_to_convert_to_managed` to be notified when
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:75]:
```suggestion
You can listen to the `access_code.unmanaged.converted_to_managed` and `access_code.unmanaged.failed_to_convert_to_managed` events to be notified when
```

File: docs/device-guides/migrating-existing-codes.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+conversion succeeds or fails.
+
+Once an unmanaged code is successfully converted to a managed code, all operations available on access codes created within
+Seam are available.
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:79]:
```suggestion
Once an unmanaged code is successfully converted to a managed code, all of the normal managed access code operations and behavior is available.
```

File: docs/device-guides/migrating-existing-codes.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+# Unsupported conversion
+
+Some third-party device providers strictly limit the information available about existing access codes on a device.
+That means that for devices connected using these providers, we cannot convert unmanaged codes to managed ones. The current
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:84]:
```suggestion
Thus, for devices connected using these providers, we cannot convert unmanaged codes to managed ones. The current
```

File: docs/device-guides/migrating-existing-codes.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+list of providers which do not support conversion are:
+
+- SmartThings
+- igloohome
+
+For SmartThings devices, you can still see a list of all unmanaged codes on the devices using the `GET /access_codes/unmanaged/list`
+endpoint but you will not be able to see the actual pin code for the access codes (the `code` property won't be present).
```
Comment (razor-x) [inline docs/device-guides/migrating-existing-codes.md:91]:
```suggestion
endpoint, but you will not be able to see the actual pin code for the access codes (the `code` property won't be present).
```

**Review by razor-x — APPROVED** [review]
This is great! Just some small copy edit suggestions.

**Review by seveibar — APPROVED** [review]
Nice, awesome suggestions from @razor-x as well

File: docs/device-guides/migrating-existing-codes.md (hunk 5 above)
Comment (rchodava) [reply docs/device-guides/migrating-existing-codes.md:21]: didn't want to add this yet until we support it, will add it in then

**Pull request merged**